Update build documentation (#17444)
apache/druid: docs/development/build.md

[here](https://github.com/apache/druid/blob/master/docs/development/build.md):
make sure it has `/master/` in the URL.

## Prerequisites

### Installing Java and Maven

- See our [Java documentation](../operations/java.md) for information about obtaining a supported JDK
- [Maven version 3.x](http://maven.apache.org/download.cgi)

### Other Dependencies

- Distribution builds require Python 3.x and the `pyyaml` module.
- Integration tests require `pyyaml` version 5.1 or later.

## Downloading the Source Code

```bash
git clone git@github.com:apache/druid.git
cd druid
```

## Building from Source

The basic command to build Druid from source is:

```bash
mvn clean install
```

> This will run static analysis, unit tests, compile classes, and package the projects into JARs. It will _not_ generate the source or binary distribution tarball. Take note that this build will take up to an hour to complete.

Reviewer: Consider something like this, as the build and test time will only increase as we add more tests, extensions, etc:
```suggestion
This will run static analysis, unit tests, compile classes, and package the projects into JARs. It will _not_ generate the source or binary distribution tarball. Note that this build may take some time to complete.
```

mvn clean install -Papache-release,dist,rat -DskipTests
```

### Building for Development

> If you are building the project only for development, you will not need the Apache release and Apache Rat profile. 

Reviewer: ```suggestion
For development, use only the dist profile and skip the Apache release and Apache rat profiles.
```

```bash
mvn clean install -Pdist -DskipTests
```

> Should you want to further speed up the build, you can enable parallel building with the `-T1C` option, and exclude some static analysis checks to further speed up the build:

Reviewer: ```suggestion
If you want to speed up the build even more, you can enable parallel building with the `-T1C` option and skip some static analysis checks.
```
